framework/src/kernels/ADTimeDerivative.C

```diff
@@ -0,0 +1,31 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADTimeDerivative.h"
+
+registerADMooseObject("MooseApp", ADTimeDerivative);
+
+defineADValidParams(
+    ADTimeDerivative,
+    ADTimeKernel,
+    params.addClassDescription("The time derivative operator with the weak form of $(\\psi_i, "
+                               "\\frac{\\partial u_h}{\\partial t})$."););
+
+template <ComputeStage compute_stage>
+ADTimeDerivative<compute_stage>::ADTimeDerivative(const InputParameters & parameters)
+  : ADTimeKernel<compute_stage>(parameters)
+{
+}
+
+template <ComputeStage compute_stage>
+ADResidual
+ADTimeDerivative<compute_stage>::computeQpResidual()
+{
+  return _test[_i][_qp] * _u_dot[_qp];
+}
```

Comment:
Doesn't this need something like `template class ADDiffusion<RESIDUAL>;` to inherit from this?

Reply:
Yep!

Reply:
Sigh, I just pushed haha.

Reply:
Ah, alright. It was down to the line on that one!